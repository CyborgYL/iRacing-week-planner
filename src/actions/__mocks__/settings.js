import debounce from 'lodash.debounce';

export const getSettingsFromFirebase = jest.fn(() => ({ type: 'SETTINGS/LOAD_SETTINGS_FROM_FIREBASE' }));
export const saveSettingsToFirebase = jest.fn(() => ({ type: 'SETTINGS/FIREBASE_SYNCED' }));

const dispatchSaveSettingsToFirebase = (dispatch) => {
  dispatch(saveSettingsToFirebase());
};

export const debouncedDispatcherSaveSettings = debounce(dispatchSaveSettingsToFirebase, 10000);
