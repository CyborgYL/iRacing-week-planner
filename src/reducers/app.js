import moment, { duration } from 'moment';
import { CHANGE_MODAL, UPDATE_DAYS } from '../actions/app';
import { SIGNED_IN } from '../actions/auth';
import { seasonStart, weekSeasonStart } from '../config';

const currentDate = moment(new Date()).utc().startOf('day');
const currentDays = currentDate.diff(seasonStart, 'days');
const getWeek = (date) => Math.ceil(duration(moment(date).add({ second: 1 }).diff(weekSeasonStart)).asWeeks());

export default function app(
  state = {
    date: currentDate,
    daysSinceSeasonStart: currentDays,
    week: getWeek(currentDate),
    currentModal: null,
  },
  { type, days, modalName },
) {
  if (type === UPDATE_DAYS) {
    const date = moment(seasonStart).add(days, 'days');
    return {
      ...state,
      date,
      daysSinceSeasonStart: days,
      week: getWeek(date),
    };
  }

  if (type === CHANGE_MODAL) {
    return {
      ...state,
      currentModal: modalName,
    };
  }

  if (type === SIGNED_IN) {
    return {
      ...state,
      currentModal: null,
    };
  }

  return state;
}
