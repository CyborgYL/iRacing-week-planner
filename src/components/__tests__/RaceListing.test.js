import { describe, test, beforeEach, afterEach } from '@jest/globals';
import moment from 'moment';
import React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate';

import RaceListing from '../RaceListing';
import { defaultFilters } from '../../reducers/settings';

import '../../data/season.json';

jest.mock('../../data/season.json');

describe('components/RaceListing', () => {
  const defaultProps = {
    date: moment('2020-09-23T01:00:00.000Z'),
    sort: { key: 'id', order: 'asc' },
    filters: defaultFilters,
    favouriteSeries: [],
    ownedTracks: [],
    ownedCars: [],
    favouriteCars: [],
    favouriteTracks: [],
    columnIds: ['id'],
    updateSort: () => {},
  };

  beforeEach(() => {
    MockDate.set('2020-09-23T13:30:00.000Z');
  });

  afterEach(() => {
    MockDate.reset();
  });

  test('renders correctly', () => {
    const component = shallow(<RaceListing
      {...defaultProps}
    />);

    expect(component).toMatchSnapshot();

    component.find({ id: 'raceListing-th-id' }).first().simulate('click');
  });

  test('changes sort on click', () => {
    const sort = jest.fn();
    const component = shallow(<RaceListing
      {...defaultProps}
      sort={{ key: 'id', order: 'asc' }}
      columnIds={['id', 'series']}
      updateSort={sort}
    />);

    expect(component).toMatchSnapshot();

    component.find({ id: 'raceListing-th-id' }).first().simulate('click');
    expect(sort).toBeCalledWith({ key: 'id', order: 'desc' });

    sort.mockClear();

    component.find({ id: 'raceListing-th-series' }).first().simulate('click');
    expect(sort).toBeCalledWith({ key: 'series', order: 'asc' });

    const newSort = jest.fn();
    const componentDesc = shallow(<RaceListing
      {...defaultProps}
      sort={{ key: 'series', order: 'desc' }}
      columnIds={['id', 'series', 'raceTimes']}
      updateSort={newSort}
    />);

    expect(componentDesc).toMatchSnapshot();

    componentDesc.find({ id: 'raceListing-th-series' }).first().simulate('click');
    expect(newSort).toBeCalledWith({ key: 'series', order: 'asc' });

    newSort.mockClear();

    componentDesc.find({ id: 'raceListing-th-raceTimes' }).first().simulate('click');
    expect(newSort).not.toBeCalled();
  });

  test('renders not found favourite series', () => {
    const component = shallow(<RaceListing
      {...defaultProps}
      filters={{ ...defaultFilters, favouriteSeries: true }}
      favouriteSeries={[543]}
    />);

    expect(component).toMatchSnapshot();
    expect(component.find('table').length).toBe(0);
  });

  test('renders not found favourite cars', () => {
    const component = shallow(<RaceListing
      {...defaultProps}
      filters={{ ...defaultFilters, favouriteCarsOnly: true }}
      favouriteCars={[9999]}
    />);

    expect(component).toMatchSnapshot();
    expect(component.find('table').length).toBe(0);
  });

  test('renders not found favourite tracks', () => {
    const component = shallow(<RaceListing
      {...defaultProps}
      filters={{ ...defaultFilters, favouriteTracksOnly: true }}
      favouriteTracks={[9999]}
    />);

    expect(component).toMatchSnapshot();
    expect(component.find('table').length).toBe(0);
  });
});
