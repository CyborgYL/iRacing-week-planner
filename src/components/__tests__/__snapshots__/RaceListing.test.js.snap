// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RaceListing changes sort on click 1`] = `
<div
  className="table-responsive raceListing"
>
  <table
    className="table"
    style={
      Object {
        "fontSize": "0.8em",
      }
    }
  >
    <thead>
      <tr>
        <th
          className="clickableCell"
          id="raceListing-th-id"
          key="id"
          onClick={[Function]}
        >
          ID
          <span>
             
          </span>
          <SortArrow
            sort={
              Object {
                "key": "id",
                "order": "asc",
              }
            }
          />
        </th>
        <th
          className="clickableCell"
          id="raceListing-th-series"
          key="series"
          onClick={[Function]}
        >
          Series
          <span>
             
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="139"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "MX5 Cup 2016",
              ],
              "carIds": Array [
                10394,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "P0D",
              "seasonId": 2915,
              "series": "Fanatec Global Mazda MX-5 Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 139,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Okayama International Circuit - Full Course",
              "trackId": 109,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "MX5 Cup 2016",
              ],
              "carIds": Array [
                10394,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "P0D",
              "seasonId": 2915,
              "series": "Fanatec Global Mazda MX-5 Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 139,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Okayama International Circuit - Full Course",
              "trackId": 109,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="278"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "GT1 Class",
                "Ford GT",
              ],
              "carIds": Array [
                10113,
                10389,
                10281,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT2H",
              "fixed": false,
              "licenceClass": "C",
              "licenceClassNumber": 2,
              "licenceLevel": 8,
              "official": true,
              "offset": "PT1H45M",
              "seasonId": 2948,
              "series": "Fanatec GT Challenge",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 278,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Nürburgring Combined - Gesamtstrecke 24h",
              "trackId": 186,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "GT1 Class",
                "Ford GT",
              ],
              "carIds": Array [
                10113,
                10389,
                10281,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT2H",
              "fixed": false,
              "licenceClass": "C",
              "licenceClassNumber": 2,
              "licenceLevel": 8,
              "official": true,
              "offset": "PT1H45M",
              "seasonId": 2948,
              "series": "Fanatec GT Challenge",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 278,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Nürburgring Combined - Gesamtstrecke 24h",
              "trackId": 186,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="32"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "Legends",
              ],
              "carIds": Array [
                10008,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": false,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "PT30M",
              "seasonId": 2917,
              "series": "iRacing Advanced Legends Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 32,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Oxford Plains Speedway",
              "trackId": 10,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "Legends",
              ],
              "carIds": Array [
                10008,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": false,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "PT30M",
              "seasonId": 2917,
              "series": "iRacing Advanced Legends Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 32,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Oxford Plains Speedway",
              "trackId": 10,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="167"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "K&N Series",
              ],
              "carIds": Array [
                10088,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "D",
              "licenceClassNumber": 1,
              "licenceLevel": 4,
              "official": true,
              "offset": "PT45M",
              "seasonId": 2923,
              "series": "iRacing ARCA Menards Series",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 167,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Lanier National Speedway - Asphalt",
              "trackId": 8,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "K&N Series",
              ],
              "carIds": Array [
                10088,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "D",
              "licenceClassNumber": 1,
              "licenceLevel": 4,
              "official": true,
              "offset": "PT45M",
              "seasonId": 2923,
              "series": "iRacing ARCA Menards Series",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 167,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Lanier National Speedway - Asphalt",
              "trackId": 8,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/RaceListing changes sort on click 2`] = `
<div
  className="table-responsive raceListing"
>
  <table
    className="table"
    style={
      Object {
        "fontSize": "0.8em",
      }
    }
  >
    <thead>
      <tr>
        <th
          className="clickableCell"
          id="raceListing-th-id"
          key="id"
          onClick={[Function]}
        >
          ID
          <span>
             
          </span>
        </th>
        <th
          className="clickableCell"
          id="raceListing-th-series"
          key="series"
          onClick={[Function]}
        >
          Series
          <span>
             
          </span>
          <SortArrow
            sort={
              Object {
                "key": "series",
                "order": "desc",
              }
            }
          />
        </th>
        <th
          className={null}
          id="raceListing-th-raceTimes"
          key="raceTimes"
          onClick={[Function]}
        >
          Race times
          <span>
             
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="167"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "K&N Series",
              ],
              "carIds": Array [
                10088,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "D",
              "licenceClassNumber": 1,
              "licenceLevel": 4,
              "official": true,
              "offset": "PT45M",
              "seasonId": 2923,
              "series": "iRacing ARCA Menards Series",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 167,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Lanier National Speedway - Asphalt",
              "trackId": 8,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "K&N Series",
              ],
              "carIds": Array [
                10088,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "D",
              "licenceClassNumber": 1,
              "licenceLevel": 4,
              "official": true,
              "offset": "PT45M",
              "seasonId": 2923,
              "series": "iRacing ARCA Menards Series",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 167,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Lanier National Speedway - Asphalt",
              "trackId": 8,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <NextRace
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="raceTimes"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "K&N Series",
              ],
              "carIds": Array [
                10088,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "D",
              "licenceClassNumber": 1,
              "licenceLevel": 4,
              "official": true,
              "offset": "PT45M",
              "seasonId": 2923,
              "series": "iRacing ARCA Menards Series",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 167,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Lanier National Speedway - Asphalt",
              "trackId": 8,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="32"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "Legends",
              ],
              "carIds": Array [
                10008,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": false,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "PT30M",
              "seasonId": 2917,
              "series": "iRacing Advanced Legends Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 32,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Oxford Plains Speedway",
              "trackId": 10,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "Legends",
              ],
              "carIds": Array [
                10008,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": false,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "PT30M",
              "seasonId": 2917,
              "series": "iRacing Advanced Legends Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 32,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Oxford Plains Speedway",
              "trackId": 10,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <NextRace
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="raceTimes"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "Legends",
              ],
              "carIds": Array [
                10008,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": false,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "PT30M",
              "seasonId": 2917,
              "series": "iRacing Advanced Legends Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 32,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Oxford Plains Speedway",
              "trackId": 10,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="278"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "GT1 Class",
                "Ford GT",
              ],
              "carIds": Array [
                10113,
                10389,
                10281,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT2H",
              "fixed": false,
              "licenceClass": "C",
              "licenceClassNumber": 2,
              "licenceLevel": 8,
              "official": true,
              "offset": "PT1H45M",
              "seasonId": 2948,
              "series": "Fanatec GT Challenge",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 278,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Nürburgring Combined - Gesamtstrecke 24h",
              "trackId": 186,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "GT1 Class",
                "Ford GT",
              ],
              "carIds": Array [
                10113,
                10389,
                10281,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT2H",
              "fixed": false,
              "licenceClass": "C",
              "licenceClassNumber": 2,
              "licenceLevel": 8,
              "official": true,
              "offset": "PT1H45M",
              "seasonId": 2948,
              "series": "Fanatec GT Challenge",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 278,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Nürburgring Combined - Gesamtstrecke 24h",
              "trackId": 186,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <NextRace
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="raceTimes"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "GT1 Class",
                "Ford GT",
              ],
              "carIds": Array [
                10113,
                10389,
                10281,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT2H",
              "fixed": false,
              "licenceClass": "C",
              "licenceClassNumber": 2,
              "licenceLevel": 8,
              "official": true,
              "offset": "PT1H45M",
              "seasonId": 2948,
              "series": "Fanatec GT Challenge",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 278,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Nürburgring Combined - Gesamtstrecke 24h",
              "trackId": 186,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="139"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "MX5 Cup 2016",
              ],
              "carIds": Array [
                10394,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "P0D",
              "seasonId": 2915,
              "series": "Fanatec Global Mazda MX-5 Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 139,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Okayama International Circuit - Full Course",
              "trackId": 109,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <Series
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="series"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "MX5 Cup 2016",
              ],
              "carIds": Array [
                10394,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "P0D",
              "seasonId": 2915,
              "series": "Fanatec Global Mazda MX-5 Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 139,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Okayama International Circuit - Full Course",
              "trackId": 109,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
        <NextRace
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="raceTimes"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "MX5 Cup 2016",
              ],
              "carIds": Array [
                10394,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "P0D",
              "seasonId": 2915,
              "series": "Fanatec Global Mazda MX-5 Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 139,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Okayama International Circuit - Full Course",
              "trackId": 109,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/RaceListing renders correctly 1`] = `
<div
  className="table-responsive raceListing"
>
  <table
    className="table"
    style={
      Object {
        "fontSize": "0.8em",
      }
    }
  >
    <thead>
      <tr>
        <th
          className="clickableCell"
          id="raceListing-th-id"
          key="id"
          onClick={[Function]}
        >
          ID
          <span>
             
          </span>
          <SortArrow
            sort={
              Object {
                "key": "id",
                "order": "asc",
              }
            }
          />
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="139"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "MX5 Cup 2016",
              ],
              "carIds": Array [
                10394,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "P0D",
              "seasonId": 2915,
              "series": "Fanatec Global Mazda MX-5 Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 139,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Okayama International Circuit - Full Course",
              "trackId": 109,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="278"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "GT1 Class",
                "Ford GT",
              ],
              "carIds": Array [
                10113,
                10389,
                10281,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT2H",
              "fixed": false,
              "licenceClass": "C",
              "licenceClassNumber": 2,
              "licenceLevel": 8,
              "official": true,
              "offset": "PT1H45M",
              "seasonId": 2948,
              "series": "Fanatec GT Challenge",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 278,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Nürburgring Combined - Gesamtstrecke 24h",
              "trackId": 186,
              "type": "Road",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="32"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "Legends",
              ],
              "carIds": Array [
                10008,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": false,
              "licenceClass": "R",
              "licenceClassNumber": 0,
              "licenceLevel": 1,
              "official": true,
              "offset": "PT30M",
              "seasonId": 2917,
              "series": "iRacing Advanced Legends Cup",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 32,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Oxford Plains Speedway",
              "trackId": 10,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
      <tr
        key="167"
      >
        <Id
          favouriteCars={Array []}
          favouriteSeries={Array []}
          favouriteTracks={Array []}
          key="id"
          ownedCars={Array []}
          ownedTracks={Array []}
          race={
            Object {
              "carClasses": Array [
                "K&N Series",
              ],
              "carIds": Array [
                10088,
              ],
              "endTime": "2020-09-29T00:00:00.000Z",
              "everyTime": "PT1H",
              "fixed": true,
              "licenceClass": "D",
              "licenceClassNumber": 1,
              "licenceLevel": 4,
              "official": true,
              "offset": "PT45M",
              "seasonId": 2923,
              "series": "iRacing ARCA Menards Series",
              "seriesEnd": "2020-12-08T00:00:00.000Z",
              "seriesId": 167,
              "seriesStart": "2020-09-15T00:00:00.000Z",
              "setTimes": undefined,
              "startTime": "2020-09-22T00:00:00.000Z",
              "track": "Lanier National Speedway - Asphalt",
              "trackId": 8,
              "type": "Oval",
              "week": 1,
              "weekLength": "PT168H",
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/RaceListing renders not found favourite cars 1`] = `
<p>
  No races this week match your favourite cars. Try turning the filter off or adding some.
</p>
`;

exports[`components/RaceListing renders not found favourite series 1`] = `
<p>
  No races this week match your favourite series. Try turning the filter off or adding some.
</p>
`;

exports[`components/RaceListing renders not found favourite tracks 1`] = `
<p>
  No races this week match your favourite tracks. Try turning the filter off or adding some.
</p>
`;
